src/db/models/user.model.js

```diff
@@ -129,11 +142,90 @@ module.exports = (sequelize, DataTypes) => {
         });
     };
 
+    User.verifyAccountToken = function verifyAccountToken(email, expTime) {
+        // This expects and email, and an expiration time window for storing a
+        // messaging protocol `messagingProtocol` token, auth expiration, and
+        // provider for a user. Users will be asked to validate this token against
+        // the submission of `verifyMessagingProtcolToken` (With or without
+        // credentials)
+
+        // QUESTION: Do we want to add some form of rate limiting in this flow?
+        return this.find({
+            where: {
+                email,
+            },
+        })
+        .then((user) => {
+            if (!user) {
+                const err = new Error('Email not found');
+                return sequelize.Promise.reject(err);
+            }
+            // JRB removed protocol, because we can confirm this. Put back it leads to errors.
+            // JRB: This is new
+            return user.updateVerifyAccountToken(email, expTime);
+        });
+    };
+
+    User.getVerifyingUser = function getVerifyingUser(token) {
+        // This expects a `contactMethodVerificationToken`, and provides a user's username.
+        return this.findOne({
+            where: {
+                contactMethodVerificationToken: token,
+            },
+        })
+        .then((user) => {
+            if (!user) {
+                const err = new Error('Token not found');
+                return sequelize.Promise.reject(err);
+            }
+            return user.username;
+        });
+    };
+
+    User.verifyUserAccount = function verifyUserAccount(token) {
+        return this.find({
+            where: {
+                contactMethodVerificationToken: token,
+            },
+        })
+        .then((user) => {
+            if (!user) {
+                const err = new Error('Token not found');
+                return sequelize.Promise.reject(err);
+            }
+            return user.addVerifiedContact(user.contactMethodToVerify);
+        });
+    };
+
+    User.secureVerifyUserAccount = function secureVerifyUserAccount(token, password) {
+        return this.find({
+            where: {
+                contactMethodVerificationToken: token,
+            },
+        })
+        .then((user) => {
+            if (!user) {
+                const err = new Error('Token not found');
+                return sequelize.Promise.reject(err);
```

Comment:
Is there a reason you are using `sequelize.Promise` here?

Reply:
I don't think so

Reply:
modeled after other places where this is being returned in the same manner